electrum/lnaddr.py

```diff
@@ -178,6 +178,9 @@ def pull_tagged(stream):
     return (CHARSET[tag], stream.read(length * 5), stream)
 
 def lnencode(addr: 'LnAddr', privkey) -> str:
+    # see https://github.com/lightningnetwork/lightning-rfc/pull/844
+    if constants.net.NET_NAME == "signet":
+        addr.currency += "s"
```

Comment:
I don't like this. I imagine if you call `lnencode` multiple times with the same LnAddr, it only works on the first call? As it mutates the LnAddr. I haven't tested though.
The root problem here is that `net.SEGWIT_HRP` is used for both the bech32 address prefix, and the bolt11 invoice prefix; and this duplication is not true anymore for signet. Let us simply have two constants in `net` then:
`net.SEGWIT_HRP` and `net.BOLT11_HRP`.
For most networks, they would be equal, but for signet they would be different.

Reply:
As far as I've tested it was fine, but it's certainly true. I fixed it.